### PR TITLE
Feat/bac 70 build update channel endpoint

### PR DESCRIPTION
**Pull request by nodebe**
- Gets the required data from the client-side(room_id, org_id, member_id, data to update)
- Checks if member_id is a member of room
- Checks if member_id is an admin to update room
- Updates room and returns updated data

[Link to Linear Issue](https://linear.app/team-bevel/issue/BAC-70/build-an-endpoint-to-update-channel-zuri-chat-task)

### CODE REVIEW

File: backend/endpoints/rooms.py [hunk 1]
```diff
@@ -435,3 +435,76 @@ async def get_members(org_id: str, room_id: str):
...
+    )
+
+    if updater.get("role") != Role.ADMIN:
+        raise HTTPException(
+        status_code=status.HTTP_401_UNAUTHORIZED,
+        detail="must be an admin to update room.",
+    )
+
+    # Insert members into data request to avoid deletion of all members
+    data['members'] = members
```
Comment (mikengr) [inline backend/endpoints/rooms.py:491]: Why do you need this? A patch update only modifies specified fields. How do you mean to avoid deletion of all members?

**Review by mikengr — COMMENTED** [review]
@nodebe good work. I left a comment

File: backend/endpoints/rooms.py (hunk 1 above)
Comment (nodebe) [reply backend/endpoints/rooms.py:491]: For the MongoDB collection, passing in a Json makes it easier to update. If the line of code is not there, the Json will be used to replace the current on it. 

It's easier to request all and update straight up.

Comment (mikengr) [reply backend/endpoints/rooms.py:491]: I still don't understand your explanation. @nodebe

Comment (nodebe) [reply backend/endpoints/rooms.py:491]: Okay. 
The document is in Json format. It is easier to pass it generally than individually. 

During the edit, members will not be edited so it's not necessary to pass the data along. 

That's why I fetched the members and reattached to the data after it was passed.

File: backend/endpoints/rooms.py [hunk 2]
```diff
@@ -435,3 +435,76 @@ async def get_members(org_id: str, room_id: str):
...
+
+    # updates the room data in the db collection
+    update_response = await DB.update(
+        settings.ROOM_COLLECTION, document_id=room_id, data=data
+    )  
+
+    if update_response and update_response.get("status_code", None) is None:
+        room_id = {"room_id": update_response.get("data").get("object_id")}
+
```
Comment (FlorenceEgwu) [inline backend/endpoints/rooms.py]: why do you need to change the room id. shouldn't it be the same as the one originally fetched?

**Review by FlorenceEgwu — APPROVED** [review]

File: backend/endpoints/rooms.py (hunk 2 above)
Comment (nodebe) [reply backend/endpoints/rooms.py]: That's a honest mistake. I'm suppose to remove that line. Thank you for pointing it out
